source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -141,9 +140,9 @@ void AuthenticatorImpl::startVerify() {
 
   jwt_ = std::make_unique<::google::jwt_verify::Jwt>();
   ENVOY_LOG(debug, "{}: Parse Jwt {}", name(), curr_token_->token());
-  Status status = jwt_->parseFromString(curr_token_->token());
-  if (status != Status::Ok) {
-    doneWithStatus(status);
+  jwks_data_->addTokenResult(curr_token_->token(), *jwt_);
```

Comment:
We only use token cache when provider_ is specified so token parsing can be saved in the cache:
```
if (provider_) {
   jwks_data_ = jwks_cache_.findByProvider(provider_.value();

   // check token cache, lookup function should verify expiration time
   // return null if cache miss or expired
   const TokenResult* result = jwks_data_->lookupTokenCache(curr_token_);
   if (result) {
       // handle cache hit, use the cache result
       return;
   }
}